blosc/schunk.c

```diff
@@ -214,8 +258,12 @@ int blosc2_schunk_insert_chunk(blosc2_schunk *schunk, int nchunk, uint8_t *chunk
   int32_t nbytes = sw32_(chunk + 4);
   int32_t cbytes = sw32_(chunk + 12);
 
-  if ((schunk->nchunks > 0) && (nbytes > schunk->chunksize)) {
-    fprintf(stderr, "appending chunks with a larger chunksize than schunk is not allowed yet: "
+  if (schunk->chunksize == -1) {
+    schunk->chunksize = nbytes;  // Only update chunksize when it is the first chunk
+  }
+
+  if (nbytes > schunk->chunksize) {
+    fprintf(stderr, "inserting chunks with a larger chunksize than schunk is not allowed yet: "
```

Comment:
Suggestion: "Inserting chunks that have different lengths in the same schunk is not supported yet."